### PR TITLE
MRG: Allow butterfly parameter of Report.add_raw() to accept number of segments to plot

### PR DESCRIPTION
**Pull request by hoechenberger**
The parameter `butterfly` allows users to specify how many 1-second segments to display as butterfly plots. We keep the previous default number (10 segments).

Performance implications:
@agramfort and I were struggling with the long run time on `main`, so I decided to reduce the number of plotted segments to 50% for our specific use case.

```
10 segments: 12.1 s ± 761 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
 5 segments: 6.29 s ± 132 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
MWE:
```python
# %%
from pathlib import Path
import mne


sample_dir = Path(mne.datasets.sample.data_path())
sample_fname = sample_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'

raw = mne.io.read_raw_fif(sample_fname, preload=True)
raw.crop(tmax=60)
r = mne.Report()

# %%
%timeit r.add_raw(raw=raw, title='foo', butterfly=10)

# %%
%timeit r.add_raw(raw=raw, title='foo', butterfly=5)
```

x-ref #10107

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
so far so good 🙏

**Comment by hoechenberger** [comment]
CI failure seems unrelated.

**Comment by hoechenberger** [comment]
Theoretically, we could deprecate the `butterfly` parameter now – to disable butterfly plots, users could set `n_butterfly_segments=0`. @agramfort WDYT?

**Comment by agramfort** [comment]
I would prefer to just accept an int for the butterfly parameter so we don't have to go for a deprecation cycle...
if it's a bool True means maybe 5 segments but you can set butterfly=1 to get only one segment.

**Comment by hoechenberger** [comment]
@agramfort Back to using only one parameter (`butterfly`) which now accepts a bool and an int, like proposed

**Review by agramfort — APPROVED** [review]
thx @hoechenberger

File: mne/report/report.py [hunk 1]
```diff
@@ -995,9 +996,12 @@ def add_raw(self, raw, title, *, psd=None, projs=None, butterfly=True,
...
         %(report_projs)s
-        butterfly : bool
-            Whether to add a butterfly plot of the (decimated) data. Can be
-            useful to spot segments marked as "bad" and problematic channels.
+        butterfly : bool | int
+            Whether to add butterfly plots of the data. Can be useful to
+            spot problematic channels. If ``True``, 10 equally-spaced 1-second
+            segments will be plotted. If an integer, specifies the number of
+            1-second segments to plot. Larger numbers may take a considerable
+            amount of time if the data contains many sensors.
```
Comment (hoechenberger) [inline mne/report/report.py:1004]: Add behavior for False

**Comment by agramfort** [comment]
thx @hoechenberger

**Pull request merged**